lib/presenters/properties_presenter.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# presents properties as: Array[ [roperty_id, property_title, property_status] ]
+
+class PropertiesPresenter
+  def initialize(properties, booking_events_by_property_id = {})
+    @properties = properties
+    @booking_events_by_property_id = booking_events_by_property_id
+  end
+
+  def present
+    @properties.map do |property|
+      [property.id, property.title, property_status(booking_events_by_property_id[property.id]&.last)]
+    end
+  end
+
+  private
+
+  def property_status(last_booking_event)
+    return 'empty' if last_booking_event.nil? || last_booking_event.type == 'BookingCheckedOut'
```

Comment:
Не уверен, но вроде будет тоже самое:
```suggestion
    return 'empty' unless last_booking_event&.type != 'BookingCheckedOut'
```